src/components/Root/Root.tsx

```diff
@@ -1,77 +1,65 @@
 import DemoConfigDialog from '#components/DemoConfigDialog/DemoConfigDialog';
-import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
 import DevConfigSelector from '#components/DevConfigSelector/DevConfigSelector';
-import { useConfigSource } from '#src/utils/configOverride';
-import { loadAndValidateConfig } from '#src/utils/configLoad';
-import { initSettings } from '#src/stores/SettingsController';
 import AppRoutes from '#src/containers/AppRoutes/AppRoutes';
 import registerCustomScreens from '#src/screenMapping';
+import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
+import { useBootstrapApp, type BootstrapData } from '#src/hooks/useBootstrapApp';
 
-// This is moved to a separate, parallel component to reduce rerenders
-const RootLoader = ({ setAppIsReady }: { setAppIsReady: React.Dispatch<React.SetStateAction<boolean>> }) => {
-  const { t } = useTranslation('error');
-  const settingsQuery = useQuery('settings-init', initSettings, {
-    enabled: true,
-    retry: 1,
-    refetchInterval: false,
-  });
-
-  const configSource = useConfigSource(settingsQuery?.data);
-
-  const configQuery = useQuery('config-init-' + configSource, async () => await loadAndValidateConfig(configSource), {
-    enabled: settingsQuery.isSuccess,
-    retry: configSource ? 1 : 0,
-    refetchInterval: false,
-  });
-
-  // After the config loads, we can show the rest of the App
-  useEffect(() => {
-    setAppIsReady(!configQuery.isError && !configQuery.isLoading && !!configQuery.data);
-  }, [setAppIsReady, configQuery.isError, configQuery.isLoading, configQuery.data]);
+const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
+
+const ProdContentLoader = ({ query }: { query: BootstrapData }) => {
+  const { isLoading, error } = query;
 
-  const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
+  if (isLoading) {
+    return <LoadingOverlay />;
+  }
+
+  if (error) {
+    return <ErrorPage title={error.payload.title} message={error.payload.description} helpLink={error.payload.helpLink} />;
+  }
+
+  return null;
+};
+
+const DemoContentLoader = ({ query }: { query: BootstrapData }) => {
+  const { isLoading, error, data } = query;
 
   // Show the spinner while loading except in demo mode (the demo config shows its own loading status)
-  if (settingsQuery.isLoading || (!IS_DEMO_OR_PREVIEW && configQuery.isLoading)) {
+  if (!IS_DEMO_OR_PREVIEW && isLoading) {
     return <LoadingOverlay />;
   }
 
-  if (settingsQuery.isError) {
-    return (
-      <ErrorPage
-        title={t('settings_invalid')}
-        message={t('check_your_settings')}
-        error={settingsQuery.error as Error}
-        helpLink={'https://github.com/jwplayer/ott-web-app/blob/develop/docs/initialization-file.md'}
-      />
-    );
+  if (error && !data?.settings) {
+    return <ErrorPage title={error.payload.title} message={error.payload.description} error={error} helpLink={error.payload.helpLink} />;
   }
```

Comment:
I don't think you need a separate return for this. I think you can just add a check for `settings` below.
```suggestion
```